packages/interactivity/src/test/utils.ts

```diff
@@ -25,4 +27,256 @@ describe( 'Interactivity API', () => {
 			expect( kebabToCamelCase( '-my-item-' ) ).toBe( 'myItem' );
 		} );
 	} );
+
+	describe( 'withScope', () => {
+		const previousNamespace = 'previousNamespace';
+		const previousScope = {
+			evaluate: () => {},
+			context: {},
+			serverContext: {},
+			ref: { current: null },
+			attributes: {},
+		};
+
+		const dummyNamespace = 'testNamespace';
+		const dummyScope = {
+			evaluate: () => {},
+			context: { test: 'normal' },
+			serverContext: { test: 'normal' },
+			ref: { current: null },
+			attributes: {},
+		};
+
+		// Helper to mimic the wrapper used in state-proxy.ts and store-proxy.ts.
+		const dummyScopeAndNS = ( callback: () => any ) => {
+			setScope( dummyScope );
+			setNamespace( dummyNamespace );
+			try {
+				return callback();
+			} finally {
+				resetNamespace();
+				resetScope();
+			}
+		};
+
+		beforeEach( () => {
+			setScope( previousScope );
+			setNamespace( previousNamespace );
+		} );
+
+		afterEach( () => {
+			resetNamespace();
+			resetScope();
+		} );
+
+		it( 'should return a function when passed a normal function', () => {
+			function normalFn( x: number ) {
+				return x + 1;
+			}
+			const wrapped = withScope( normalFn );
+			expect( typeof wrapped ).toBe( 'function' );
+		} );
+
+		it( 'should call the original normal function', () => {
```

Comment:
This test is redundant with the following one (`should set the scope and namespace and restore them afterwards`).